### PR TITLE
[#3906] improvement(hive-catalog): Add user authentication IT for multiple HMS

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add IT about user authentication for Hive catalog with Kerberos enabled.

### Why are the changes needed?

To cover more usage cases. 

Fix: #3906 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by yuqi1129** [comment]
@mchades 
Would you be able to help me review this PR if you have some free time?

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/MultipleHMSUserAuthenticationIT.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.junit.jupiter.api.Test;
+
+@Tag("gravitino-docker-test")
+public class MultipleHMSUserAuthenticationIT extends HiveUserAuthenticationIT {
+  @BeforeAll
+  static void setHiveURI() {
+    String ip = kerberosHiveContainer.getContainerIpAddress();
+    int port = HiveContainer.HIVE_METASTORE_PORT;
+    // Multiple HMS URIs, I put the new one first to test the new HMS URI first.
+    HIVE_METASTORE_URI = String.format("thrift://%s:1%d,thrift://%s:%d", ip, port, ip, port);
```
Comment (mchades) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/MultipleHMSUserAuthenticationIT.java:35]: please use a new port parameter, `1%d` looks too tricky

Comment (yuqi1129) [reply catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/integration/test/MultipleHMSUserAuthenticationIT.java]: Fixed

**Comment by yuqi1129** [comment]
@mchades 
Please help to take a look again, thanks.

**Review by mchades — APPROVED** [review]
LGTM.

**Pull request merged**